test/muxer_test.exs

```diff
@@ -0,0 +1,241 @@
+defmodule Membrane.WebM.MuxerTest do
+  # note that input pad ids should be set to a random value:
+  # :random.uniform(1 <<< 64)
+  # here numbers are hardcoded to achieve reproducibility
+
+  use ExUnit.Case
+  use Bitwise
+
+  import Membrane.Testing.Assertions
+  alias Membrane.Testing
+  alias Membrane.{Buffer, Opus}
+
+  @fixtures_dir "./test/fixtures/"
+
+  defmodule TestPipelineOpus do
+    use Membrane.Pipeline
+
+    @impl true
+    def handle_init(options) do
+      source = %Testing.Source{
+        output: Testing.Source.output_from_buffers(options.buffers),
+        caps: %Opus{channels: 2, self_delimiting?: false}
+      }
+
+      children = [
+        source: source,
+        deserializer: Membrane.Element.IVF.Deserializer,
+        muxer: Membrane.WebM.Muxer,
+        sink: %Membrane.File.Sink{
+          location: options.output_file
+        }
+      ]
+
+      links = [
+        link(:source)
+        |> via_in(Pad.ref(:input, 17_447_232_417_024_423_937))
+        |> to(:muxer)
+        |> to(:sink)
+      ]
+
+      {{:ok, spec: %ParentSpec{children: children, links: links}}, %{}}
+    end
+  end
+
+  defmodule TestPipelineVpx do
+    use Membrane.Pipeline
+
+    @impl true
+    def handle_init(options) do
+      children = [
+        source: %Membrane.File.Source{
+          location: options.input_file
+        },
+        deserializer: Membrane.Element.IVF.Deserializer,
+        muxer: Membrane.WebM.Muxer,
+        sink: %Membrane.File.Sink{
+          location: options.output_file
+        }
+      ]
+
+      links = [
+        link(:source)
+        |> to(:deserializer)
+        |> via_in(Pad.ref(:input, 13_337_737_628_113_408_001))
+        |> to(:muxer)
+        |> to(:sink)
+      ]
+
+      {{:ok, spec: %ParentSpec{children: children, links: links}}, %{}}
+    end
+  end
+
+  defmodule Printer do
+    use Membrane.Filter
+
+    def_input_pad :input,
+      availability: :always,
+      mode: :pull,
+      demand_unit: :buffers,
+      caps: :any
+
+    def_output_pad :output,
+      availability: :always,
+      mode: :pull,
+      caps: :any
+
+    @impl true
+    def handle_init(_options) do
+      {:ok, %{cache: []}}
+    end
+
+    @impl true
+    def handle_demand(:output, size, :buffers, _context, state) do
+      {{:ok, demand: {:input, size}}, state}
+    end
+
+    @impl true
+    def handle_end_of_stream(:input, _context, state) do
+      IO.puts("Don't kill me - working. May take 10 seconds or so")
+      output = inspect(state.cache, limit: :infinity, pretty: true)
+      {{:ok, [buffer: {:output, %Buffer{payload: output}}, end_of_stream: :output]}, state}
+    end
+
+    @impl true
+    def handle_process(
+          :input,
+          %Buffer{payload: data, metadata: %{webm: %{element_name: element_name}}},
+          _context,
+          state
+        ) do
+      {{:ok, demand: {:input, 1}}, %{state | cache: [{element_name, data} | state.cache]}}
+    end
+  end
+
+  defmodule TestPipelineMany do
```

Comment:
In this file, instead of defining pipeline modules, I would just use Membrane.Testing.Pipeline